governance/xc_admin/packages/xc_admin_common/src/programs/lazer/lazer_functions.ts

```diff
@@ -148,24 +154,143 @@ export function getDownloadableConfig(config: LazerConfig): DownloadableConfig {
  * @returns Object with validation result and optional error message
  */
 export function validateUploadedConfig(
-  existingConfig: DownloadableConfig,
-  uploadedConfig: unknown,
+  existingConfig: LazerState,
+  uploadedConfig: LazerState,
   cluster: PythCluster,
 ): ValidationResult {
-  // Basic validation logic for Lazer config
   try {
+    // Basic type validation
     if (typeof uploadedConfig !== "object" || uploadedConfig === null) {
       return {
         isValid: false,
         error: "Invalid JSON format for Lazer configuration",
       };
     }
 
-    // More detailed validation would be implemented here
-    // For now, return not implemented error
+    // Cast to LazerState
+    const uploadedLazerState = uploadedConfig as LazerState;
+
+    // Check for required fields
+    if (
+      !("shardId" in uploadedLazerState) ||
+      !("feeds" in uploadedLazerState) ||
+      !("publishers" in uploadedLazerState)
+    ) {
+      return {
+        isValid: false,
+        error: "Missing required fields in Lazer configuration",
+      };
+    }
+
+    // Calculate changes between existing and uploaded config
+    const changes: Record<string, { prev?: any; new?: any }> = {};
```

Comment:
Please do not use `any` -- you can restructure this type slightly by using well known keys to be easier to type (and probably easier to use) with something along these lines:

```ts
type ChangeField<T> = { prev: T, new: T };

type Changes = {
  shard: ChangeField<Shard>;
  feeds: Record<string, ChangeField<Feed>>;
  publishers: Record<string, ChangeField<Publisher>>;
}
```